.github/workflows/ci.yml

```diff
@@ -0,0 +1,352 @@
+
+      - name: Install rust toolchain
+        working-directory: ${{ env.RUST_WORKSPACE }}
+        run: rustup toolchain install ${{ env.RUST_NIGHTLY }} --component rustfmt
+
+      - uses: Swatinem/rust-cache@d12701459954fec471b2d34cdf7ea3374b026383 # v1
+        with:
+          working-directory: ${{ env.RUST_WORKSPACE }}
+
+      - name: cargo fmt
+        working-directory: ${{ env.RUST_WORKSPACE }}
+        run: cargo +${{ env.RUST_NIGHTLY }} fmt --all -- --check
+
+  cargo-clippy:
+    needs: selection
+    if: ${{ needs.changes.outputs.rust == 'true' }}
+    runs-on: ubuntu-20.04
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      - name: Install rust toolchain
+        working-directory: ${{ env.RUST_WORKSPACE }}
+        run: rustup show
+
+      - uses: Swatinem/rust-cache@d12701459954fec471b2d34cdf7ea3374b026383 # v1
+        with:
+          working-directory: ${{ env.RUST_WORKSPACE }}
+
+      - name: cargo clippy
+        working-directory: ${{ env.RUST_WORKSPACE }}
+        run: |
+          # workaround for some code being created after rust wants to parse it
+          cargo check --quiet 2>/dev/null || :
+          cargo clippy --all-targets -- --deny warnings
+
+  cargo-test:
+    needs: selection
+    if: ${{ needs.changes.outputs.rust == 'true' }}
+    runs-on: ubuntu-20.04
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      - name: Install rust toolchain
+        working-directory: ${{ env.RUST_WORKSPACE }}
+        run: rustup show
+
+      - uses: Swatinem/rust-cache@d12701459954fec471b2d34cdf7ea3374b026383 # v1
+        with:
+          working-directory: ${{ env.RUST_WORKSPACE }}
+
+      - name: Run tests
+        working-directory: ${{ env.RUST_WORKSPACE }}
+        run: |
+          # workaround for some code being created after rust wants to parse it
+          cargo check --quiet 2>/dev/null || :
+          cargo test --all-targets --quiet
+          cargo test --doc --quiet
+
+  cargo-doc:
+    needs: selection
+    if: ${{ needs.changes.outputs.rust == 'true' }}
+    runs-on: ubuntu-20.04
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      - name: Install rust toolchain
+        working-directory: ${{ env.RUST_WORKSPACE }}
+        run: rustup show
+
+      - uses: Swatinem/rust-cache@d12701459954fec471b2d34cdf7ea3374b026383 # v1
+
+      - name: cargo doc
+        working-directory: ${{ env.RUST_WORKSPACE }}
+        run: |
+          # workaround for some code being created after rust wants to parse it
+          cargo check --quiet 2>/dev/null || :
+          cargo doc --all-features --no-deps --document-private-items
+
+    # this is an helper that needs all the real leafs of the workflow.
+    # It makes easier notify_main_failure because we only need to check
+    # for this job
```

Comment:
This can be deleted, or updated there is also no `notify_main_failure` around anymore in this script.

Reply:
Yes, Andrea told me that given that we were going to use `bors` the `notifyy_main_failure` was not needed anymore.